gcc/rust/hir/tree/rust-hir-pattern.h

```diff
@@ -1295,6 +1293,72 @@ class SlicePattern : public Pattern
   }
 };
 
+// HIR node for alternative patterns
+class AltPattern : public Pattern
+{
+  std::vector<std::unique_ptr<Pattern>> alts;
+  Location locus;
+  Analysis::NodeMapping mappings;
+
+public:
+public:
+  std::string as_string () const override;
+
+  AltPattern (Analysis::NodeMapping mappings,
+	      std::vector<std::unique_ptr<Pattern>> alts, Location locus)
+    : alts (std::move (alts)), locus (locus), mappings (mappings)
+  {}
+
+  // Copy constructor with vector clone
+  AltPattern (AltPattern const &other)
+    : locus (other.locus), mappings (other.mappings)
+  {
+    alts.reserve (other.alts.size ());
```

Comment:
Missing a call to `alt.clear()` otherwise you're going to run into some issues if you copy-assign to an `AltPattern` multiple times. This is a problem in some of our other classes as well so no worries
```suggestion
    alts.clear();
    alts.reserve (other.alts.size ());
```

Reply:
That's a copy constructor, so ```alts``` should have just been initialized as an empty vector.